keps/sig-storage/20190130-csi-raw-block-support.md

```diff
@@ -0,0 +1,82 @@
+---
+title: CSI Raw Block Volumes
+authors:
+  - "@bswartz"
+owning-sig: sig-storage
+participating-sigs:
+  - sig-storage
+reviewers:
+  - "@msau42"
+  - "@saad-ali"
+approvers:
+  - name: "@saad-ali"
+editor: TBD
+creation-date: 2019-01-30
+status: implementable
+---
+
+# CSI Raw Block Volumes
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-goals](#non-goals)
+* [Proposal](#proposal)
+* [Graduation Criteria](#graduation-criteria)
+   * [Alpha -&gt; Beta](#alpha---beta)
+   * [Beta -&gt; GA](#beta---ga)
+
+## Summary
+
+Raw block support has been added to both the core of Kubernetes (including
+API support and support in kubelet for a subset of volume types) as well as
+he CSI specification. This feature is specifically to add raw block support
+to kubelet for volumes of the "csi" type by taking advantage of the
+now-standardized CSI RPCs.
+
+## Motivation
+
+The future of persistent volume support should be mostly CSI-type volumes,
+with a few notable exceptions. It's therefore essential that CSI volumes
+can implement all of the features that Kubernetes has, including support
+for raw block volumes.
+
+### Goals
+
+CSI plugins should be able to create, publish, and attach raw block volumes
+in accordane with the raw block support APIs.
+
+At least one CSI driver capable of running gate tests must support raw block
+volumes so the feature can be regression tested.
+
+It's desirable to get several third party CSI plugins to support raw block
+volumes to confirm both that the interface is flexible enough to accomodate
+multiple implementations and that it's specific enough to not leave any
+ambiguity about how implementations should work.
+
+### Non-Goals
+
+Raw block support added in any particular CSI driver.
+
+## Proposal
+
+Modify the CSI volume package of kubelet to declare support for raw block
+volumes, and add the CSI GRPC client code to call the appropriate node RPCs
+for raw block volumes when attaching pods to volumes of that type. This
+support is controlled by a feature gate called "CSIBlockVolume".
+
+## Graduation Criteria
```

Comment:
Add a `Test Plan` section with a list of E2E tests we want for this features (plus plan for unit and integration tests).